File: backend/src/main/java/com/corusoft/ticketmanager/tickets/controllers/dtos/TicketDTO.java
package com.corusoft.ticketmanager.tickets.controllers.dtos;

import com.corusoft.ticketmanager.tickets.entities.CustomizedCategoryID;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TicketDTO {
    private Long id;

    private String name;

    @NotNull
    @PastOrPresent
    private LocalDateTime registeredAt;

    @NotNull
    @PastOrPresent
    private LocalDateTime emmitedAt;

    @NotNull
    private Float amount;

    @NotNull
    private String currency;

    @NotNull
    private String picture;

    private String store;

    private CustomizedCategoryID customizedCategoryID;

    @NotNull
    private Long creatorID;
}

File: mobile/app/src/main/java/com/corusoft/ticketmanager/backend/dtos/tickets/TicketDTO.kt
package com.corusoft.ticketmanager.backend.dtos.tickets

import java.time.LocalDateTime

data class TicketDTO(
    val id: Long,
    val name: String?,
    val registeredAt: LocalDateTime?,
    val emmitedAt: LocalDateTime?,
    val amount: Float?,
    val currency: String?,
    val picture: String?,
    val store: String?,
    val customizedCategoryID: CustomizedCategoryID?,
    val creatorID: Long?
)